apps/judicial-system/web/src/routes/Court/RestrictionCase/HearingArrangements/HearingArrangements.tsx

```diff
@@ -50,19 +46,12 @@ export const HearingArrangements: React.FC = () => {
 
   const [modalVisible, setModalVisible] = useState(false)
 
-  const router = useRouter()
-  const id = router.query.id
-
   const [initialAutoFillDone, setInitialAutoFillDone] = useState(false)
-  const {
-    updateCase,
-    autofill,
-    sendNotification,
-    isSendingNotification,
-  } = useCase()
+  const { autofill, sendNotification, isSendingNotification } = useCase()
   const { formatMessage } = useIntl()
 
   const [courtDate, setCourtDate] = useState(workingCase.courtDate)
+  const [courtDateHasChanged, setCourtDateHasChanged] = useState(false)
```

Comment:
I think that would not work because 
- we set `courtDate` when we change the court date 
- when we click "continue" we set `workingCase.courtDate` and then we show the modal

Therefore `courtDate` and `workingCase.courtDate` will always be equal when the modal is opened.